File: app/src/main/java/com/leeeyou/util/HtmlUtils.java
package com.leeeyou.util;

import java.util.List;

public class HtmlUtils {
    public static String structHtml(String oriStr, List<String> cssList) {
        StringBuilder htmlString = new StringBuilder("<html><head>");
        for (String css : cssList) {
            htmlString.append(structCssLink(css));
        }
        htmlString.append("</head><body>");
        htmlString.append("<style>img{max-width:340px !important;}</style>");
        htmlString.append(oriStr);
        htmlString.append("</body></html>");
        return htmlString.toString();
    }

    public static String structCssLink(String css) {
        return "<link type=\\\"text/css\\\" rel=\\\"stylesheet\\\" href=\\\"" + css + "\">";
    }
}

File: app/src/main/java/com/leeeyou/util/HtmlUtils.kt
package com.leeeyou.util

object HtmlUtils {
    fun structHtml(oriStr: String, cssList: List<String>): String {
        val htmlString = StringBuilder("<html><head>")
        for (css in cssList) {
            htmlString.append(structCssLink(css))
        }
        htmlString.append("</head><body>")
        htmlString.append("<style>img{max-width:340px !important;}</style>")
        htmlString.append(oriStr)
        htmlString.append("</body></html>")
        return htmlString.toString()
    }

    fun structCssLink(css: String): String {
        return "<link type=\\\"text/css\\\" rel=\\\"stylesheet\\\" href=\\\"$css\">"
    }

    fun translation(content: String): String {
        return content.replace("&lt;", "<")
                .replace("&gt;", ">")
                .replace("&amp;", "&")
                .replace("&quot;", "\"")
                .replace("&mdash;", "—")
                .replace("&copy;", "©")
    }

}
